java/arcs/core/data/Recipe.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+import arcs.core.type.Type
+
+/** Representation of recipe in an Arcs manifest. */
+data class Recipe(val name: String) {
+    /** Definition of a handle in a recipe. */
+    data class Handle(
+        val name: String,
+        val fate: Fate,
+        val storageKey: String,
```

Comment:
And we want a string for a StorageKey here, not an actual StorageKey?

Reply:
My current thinking is that we keep it as a string in the Recipe for now as it is kind of irrelevant for analysis and type inference. When we convert the resolved recipe into a Plan, we can convert it to an actual `StorageKey`.